### PR TITLE
Refactor `DesignerSerializationManager` dictionary usage to use `TryGetValue`

### PR DESCRIPTION
**Pull request by elachlan**
Refactor dictionary usage to use `TryGetValue`.

Addresses feedback from:
https://github.com/dotnet/winforms/pull/8353#issuecomment-1443195430

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8702)

### CODE REVIEW

**Comment by elachlan** [comment]
CC: @KalleOlaviNiemitalo @dreddy-work

**Comment by elachlan** [comment]
> Would this deserve a test too?

I'll look into it when I have more time. I am not sure how to go about it.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs [hunk 1]
```diff
@@ -420,9 +417,8 @@ public IDisposable CreateSession()
             {
                 if (serializers is not null)
                 {
-                    // I don't double hash here.  It will be a very rare day where multiple types of serializers will be used in the same scheme. We still handle it, but we just don't cache.
-                    serializer = serializers[objectType];
-                    if (serializer is not null && !serializerType.IsAssignableFrom(serializer.GetType()))
+                    // I don't double hash here. It will be a very rare day where multiple types of serializers will be used in the same scheme. We still handle it, but we just don't cache.
+                    if (serializers.TryGetValue(objectType, out serializer) && !serializerType.IsAssignableFrom(serializer.GetType()))
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs:421]: Is it possible for `objectType` to map to a `null` value in `serializers`? If so, we should check if `serializer` is null before calling `serializer.GetType()`

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs [hunk 2]
```diff
@@ -804,26 +800,22 @@ object IDesignerSerializationManager.CreateInstance(Type type, ICollection? argu
         /// </summary>
         string? IDesignerSerializationManager.GetName(object value)
         {
-            string? name = null;
             ArgumentNullException.ThrowIfNull(value);
 
             CheckSession();
             // Check our local nametable first
-            if (namesByInstance is not null)
+            if (namesByInstance is not null && namesByInstance.TryGetValue(value, out string? name))
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs:807]: Ditto here. If `value` can map to a null value in `namesByInstance`, we should check if `name` is not null before returning it.

**Comment by lonitra** [comment]
> > Would this deserve a test too?
> 
> I'll look into it when I have more time. I am not sure how to go about it.

Off the top of my head I am not sure how complicated our unit tests for `DesignerSerializerManager` are, but it is worth revisiting those tests to see if our existing tests that exercise these methods calling to the now dictionaries also exercises it with keys that don't exist to ensure we are not unexpectedly throwing.

**Comment by dreddy-work** [comment]
@elachlan, just checking if this is still on your radar.

**Comment by elachlan** [comment]
Yes. Work has taken over a bit due to some deadlines. I'll see what I can do.

**Comment by dreddy-work** [comment]
@elachlan, did you get chance to respond on couple of open comment above?

**Comment by elachlan** [comment]
I'll look today.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs (hunk 2 above)
Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs:807]: namesByInstance/instancesByName do not accept a nullable value.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs (hunk 1 above)
Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs:421]: We check `objectType` for null just above on line 416.

**Comment by elachlan** [comment]
@dreddy-work I believe its addressed. But feel free to add to it if I have missed something.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**